### PR TITLE
test(client): add integration tests for Java client

### PR DESCRIPTION
**Pull request by vcrfxia**
### Description 

As the title says, this PR adds integration tests for the Java client.

### Testing done 

Test-only change.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+    verifyPullQueryRow(row);
+    assertThat(streamedQueryResult.poll(), is(nullValue()));
+
+    assertThatEventually(streamedQueryResult::isComplete, is(true));
+  }
+
+  @Test
+  public void shouldStreamPushQueryWithLimitAsync() throws Exception {
+    // When
+    final StreamedQueryResult streamedQueryResult = client.streamQuery(PUSH_QUERY_WITH_LIMIT).get();
```
Comment (purplefox) [inline ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java:239]: I think the limit here is just the number of rows - would be better imho to test with limit < number of rows to make sure the limit really works.

File: ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+    assertThat(e.getCause(), instanceOf(KsqlRestClientException.class));
+    assertThat(e.getCause().getMessage(), containsString("Received 400 response from server"));
+    assertThat(e.getCause().getMessage(), containsString("No query with id NONEXISTENT"));
+  }
+
+  private Client createClient() {
+    final ClientOptions clientOptions = ClientOptions.create()
+        .setHost("localhost")
+        .setPort(REST_APP.getListeners().get(0).getPort())
+        .setUseTls(false);
```
Comment (purplefox) [inline ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java:400]: Isn't default tls false already?

File: ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java (hunk 1 above)
Comment (vcrfxia) [reply ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java:240]: Seems more like server-side functionality testing than client functionality testing but there's no harm in testing it here too so I've made the update.

File: ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java (hunk 2 above)
Comment (vcrfxia) [reply ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java]: Yeah. Removed.

**Pull request merged**